skyportal/handlers/api/internal/recent_sources.py

```diff
@@ -96,7 +96,7 @@ def get(self):
 
 
 def first_thumbnail_public_url(thumbnails):
-    urls = [t.public_url for t in sorted(thumbnails, key=lambda t: tIndex(t))]
+    urls = [t.public_url for t in sorted(thumbnails, key=lambda t: tIndex(t.type))]
```

Comment:
Is this relying on "new" being before "ref" and "sub" alphabetically? What if a "dr8" thumbnail is present?

Reply:
how about grabbing the "new" thumbnail explicitly, and then falling back to this line here if there's no thumbnail with type "new"?

Reply:
also, since this is duplicated across multiple handlers, let's just define it once and then import it in the other file

Reply:
@dannygoldstein @acrellin No, the `tIndex()` function being called there to determine the sorting is actually defined right below this, and defines the explicit ordering of thumbnail types. So it's not just alphabetical, and all the logic including falling back is already in there.
![Screenshot 2020-10-16 092457](https://user-images.githubusercontent.com/17696889/96284069-f3ce7300-0f91-11eb-92e3-7b50b66a4e8c.png)

I did the deduplication of the function though.